New Black Market Uplink Weapon - Carpenter Hammer

fevryk opened the pull request:
## About The Pull Request

There are maints hidden under every station in the world.
Everyone knew about them, but lied to you about them your entire life.

This PR adds carpenter hammer as a new black market uplink weapon with a sliiiight psychopomp reference (!)

      



![dreamseeker_9eImatpaTd](https://github.com/user-attachments/assets/796463a0-9c88-4f77-aa1a-98d079a3050f)
(code is mostly done with the help of my awesome friend Sensum)

## Why It's Good For The Game

More use for black market uplink, and new weapon with cool killsound feature.
I want black market to be more content filled with various stuff.


https://github.com/user-attachments/assets/f726d292-dc14-43b2-a536-21901bb7f9b9



![dreammaker_odBCfLOltA](https://github.com/user-attachments/assets/7cb212de-cc3b-4298-9121-d82569f455ba)

![Aseprite_eEG9E3PLlY](https://github.com/user-attachments/assets/3264733e-e58a-4c3b-b1b2-27693b73924a)

![Aseprite_kCZV4Akb3x](https://github.com/user-attachments/assets/27ae4b49-fed5-414f-9dba-90a2441835f8)


## Changelog

:cl:
add: Added new weapon carpenter hammer to black market uplink
add: Added killsound feature
/:cl:

SSensum13 committed:
The hammer

Commit:
The hammer

Commit:
Merge branch 'master' into hammer

Commit:
updated sound path

Sealed101 commented:
something something sound effect license

Niron3206 commented:
Cool!

Ghommie commented on code/datums/components/item_killsound.dm:
```diff
@@ -0,0 +1,39 @@
+/datum/component/item_killsound
```
A small doc comment introducing the component here would be nice.

Ghommie commented on code/datums/components/item_killsound.dm:
```diff
@@ -0,0 +1,39 @@
+	var/blacklisted_mobs
+	var/killsound
+	var/killsound_volume = 100
+	/// on true will act as replacement for mob's death sound,
+	/// otherwise it will just play sound on death
```
```suggestion
	/**
	 * on true will act as replacement for mob's death sound,
	 * otherwise it will just play sound on death
	 */
```
multiline docs

Ghommie commented on code/datums/components/item_killsound.dm:
```diff
@@ -0,0 +1,39 @@
+	/// list of allowed types, not null/empty
+	var/allowed_mobs
+	/// list of blacklisted types
+	var/blacklisted_mobs
```
```suggestion
	var/list/blacklisted_mobs
```

Ghommie commented on code/datums/components/item_killsound.dm:
```diff
@@ -0,0 +1,39 @@
+/datum/component/item_killsound
+	/// list of allowed types, not null/empty
+	var/allowed_mobs
```
```suggestion
	var/list/allowed_mobs
```

Ghommie commented on code/datums/status_effects/death_sound.dm:
```diff
@@ -0,0 +1,46 @@
+/datum/status_effect/death_sound
```
Doc.

Ghommie commented on sound/items/weapons/hammer_death_scream.ogg:
Is any attribution needed for this sound? Depending on where you got the file, an attribution including the original name of the sound, the author and perhaps some extra text and a link to it is needed. If you ripped it from the game, then it's a bit of a no no (unless it's CC or similar to begin with, or SOMEHOW got the permission from the owner, but even then it'd need some formal proof ig). Even if unlikely, being a fairly niche game, we don't want to deal with copyright issues.

~~I never played the game and it's been a while since I watched a playthrough of it.~~

Sadboysuss replied:
small nit for file organization - this is should be in the effects folder rather than the weapons folder, but it's not a big deal, and yeah you need attribution, preferable that you make a folder for the file and place the attribution txt together with the sound file in the folder if you end up making one.

ArcaneMusic commented on code/modules/cargo/markets/market_items/weapons.dm:
```diff
@@ -64,6 +64,16 @@
+	desc = "When you really want to look like a psycho..."
+	item = /obj/item/carpenter_hammer
+
+	price_min = CARGO_CRATE_VALUE * 1
+	price_max = CARGO_CRATE_VALUE * 1.25
+	stock_max = 2
+	availability_prob = 65
```
I have concerns about the availability of this item, as you're offering it at less than the chainsaw's cost, at higher availability, and higher stock, for an arguably higher offensive weapon. For reference, here's the chainsaw's market_item qualities.
```
	price_min = CARGO_CRATE_VALUE * 1.75 //350 credits
	price_max = CARGO_CRATE_VALUE * 3 //600 credits
	stock_max = 1
	availability_prob = 35```
	
I would want it to be either a higher price (Maybe 400-800 credits), or a lower availability/stock before I ran with this.

Ghommie replied:
I think you're overshooting it. It's better to have something that's not underpowered out of mere prudence on release. That way we get people to try it out and it gives us actual feedback.

ArcaneMusic replied:
The way it's sitting right now, it's also less expensive than every other weapon on the black market, as a 20 force, normal sized melee weapon, WITH a demolition_mod of 25%. I'm not suggesting nerfing it into the ground, just that it's starting out very high on the power scale.

Ghommie replied:
All good, I simply meant to say that the values you were suggesting seemed a smidge too tall, especially the availability.

carlarctg replied:
the stats are undertuned on every other black market item which comparatively makes this look overtuned

CCC23 commented:
https://x.com/Senn_Spud/status/1438947798469148673?lang=en

deathrobotpunch commented:
beating you to death with hammers rn

DaCoolBoss commented:
The other claw hammer functions as a crowbar (with half tool speed) so it'd be nice for consistancy if this also had crowbar tool functions. Add "tool_behaviour = TOOL_CROWBAR" to the list of variables if you want this.

It would also be cool if this was whitelisted for storage in toolboxes and toolbelts, but the toolbelt would need new sprites (current hammer toolbelt sprite is a wooden-handled one rather than rubber-handled, it's usable but it'll look noticeably off). Again this isn't major, it's just a thematic way of carrying this item around.

fevryk commented:
I'll draft the PR until we get permission to sound or we find similar one. Suggestions will be pleased once we figure out about sound and IP.

Sadboysuss commented:
If worse comes to worst and you can't find the source - just pr it without the sound :3

Loafin34 commented:
Died 2018 Born 2024 Welcome back combat knife (20 damage normal weight weapon)

carlarctg commented:
combat knives are small sized and you can already buy switchblades which are also small sized* from the same black market.

LemonInTheDark commented:
@fevryk what's goin on here?

Ghommie commented:
Yeah, don't expect a total stranger to respond to you, moreso to give you permissions. You should've chosen the other option from the get-go. Maybe you can record a few screams or have someone else do them, and then have the sounds mixed with some FFmpeg options (or Audacity or whatever you want to use).

fevryk commented:
> Yeah, don't expect a total stranger to respond to you, moreso to give you permissions. You should've chosen the other option from the get-go. Maybe you can record a few screams or have someone else do them, and then have the sounds mixed with some FFmpeg options (or Audacity or whatever you want to use).

Yes, it's my fault. But the right thoughts are not always the first. One way or another, we got the sound. (We recorded how our friend's mother sends him to the grocery store)

SSensum13 committed:
Update hammer_death_scream.ogg

grungussuss reviewed (commented):
Pls leave a note in `attribution.txt` the source of the file and the license you release it under. It's not a requirement, but it makes things so much easier for the future

Niron3206 committed:
modified:   icons/mob/inhands/weapons/hammers_lefthand.dmi
	modified:   icons/mob/inhands/weapons/hammers_righthand.dmi
	modified:   sound/attributions.txt

grungussuss reviewed (commented):
sound ***sounds*** good to me.

fevryk committed:
Apply suggestions from code review

Co-authored-by: Ghom <42542238+Ghommie@users.noreply.github.com>

Ghommie commented:
More conflicts.

fevryk commented:
> More conflicts.

arghh dammit

SSensum13 committed:
update icons before merge from master

SSensum13 committed:
Merge branch 'master' into hammer

SSensum13 committed:
fixed mistake with icon states

Fikou commented on code/game/objects/items/weaponry.dm:
```diff
@@ -432,6 +432,42 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
+		"Every building, from hospitals to homes, has a room that serves as the heart of the building \
+		and carries blood and nutrients to its extremities. Try to find the heart of your home!",
+		"All the food you've tried is rotten. You've never eaten fresh food.",
+		"Viruses do not exist. Illness is simply your body punishes you for what you have done wrong.",
```
```suggestion
		"Viruses do not exist. Illness is simply your body punishing you for what you have done wrong.",
```

fevryk committed:
Update code/game/objects/items/weaponry.dm

Co-authored-by: Fikou <23585223+Fikou@users.noreply.github.com>
